multiaddr/protocols.py

```diff
@@ -63,62 +63,64 @@
 ]
 
 
-# These are special sizes
-LENGTH_PREFIXED_VAR_SIZE = -1
-
+find_codec_by_name = None
 
 class Protocol(object):
     __slots__ = [
         "code",   # int
-        "size",   # int (-1 indicates a length-prefixed variable size)
         "name",   # string
-        "vcode",  # bytes
         "codec",  # string
-        "path",   # bool (True indicates a path protocol (eg unix, http))
     ]
 
-    def __init__(self, code, size, name, codec=None, path=False):
+    def __init__(self, code, name, codec):
         if not isinstance(code, six.integer_types):
             raise ValueError("code must be an integer")
-        if not isinstance(size, six.integer_types):
-            raise ValueError("size must be an integer")
         if not isinstance(name, six.string_types):
             raise ValueError("name must be a string")
-        if not isinstance(codec, six.string_types) and codec is not None:
+        if not isinstance(codec, six.string_types) and not codec is None:
             raise ValueError("codec must be a string or None")
-        if not isinstance(path, bool):
-            raise ValueError("path must be a boolean")
 
         if code not in _CODES and code != 0:
             raise ValueError("Invalid code '%d'" % code)
 
-        if size < -1 or size > 128:
-            raise ValueError("Invalid size")
         self.code = code
-        self.size = size
         self.name = name
-        self.vcode = varint.encode(code)
         self.codec = codec
-        self.path = path
+
+    @property
+    def size(self):
+        global find_codec_by_name
+        if find_codec_by_name is None:
+            from .codec import find_codec_by_name
+
+        return find_codec_by_name(self.codec).SIZE
+
+    @property
+    def path(self):
+        global find_codec_by_name
+        if find_codec_by_name is None:
+            from .codec import find_codec_by_name
```

Comment:
Probably this code snippet can be put to another method? It is used in `size` as well

Reply:
I've found a different solution for this: `find_codec_by_name` has been moved in the package constructor of `codecs`; thereby avoiding the import loop entirely.